### PR TITLE
Update fltk.rb to support HEAD for OS X El Capitan

### PR DESCRIPTION
**Pull request by jmtatsch**
Added HEAD option for building from svn in order to build with OS X El Capitan

### CODE REVIEW

File: Library/Formula/fltk.rb [hunk 1]
```diff
@@ -11,27 +9,42 @@ class Fltk < Formula
 
   option :universal
 
+  stable do
+    # Fltk 1.3.4 include support for El Capitan. Remove on update.
```
Comment (DomT4) [inline Library/Formula/fltk.rb]: These two lines (The comment and the depends_on) should be one empty line beneath the `:DATA` patch line.

File: Library/Formula/fltk.rb [hunk 2]
```diff
@@ -11,27 +9,42 @@ class Fltk < Formula
...
+  head do
+    url "http://seriss.com/public/fltk/fltk/branches/branch-1.3/", :using => :svn
+  end
+
   depends_on "libpng"
   depends_on "jpeg"
-
-  # Fltk 1.3.4 include support for El Capitan. Remove on update.
-  depends_on MaximumMacOSRequirement => :yosemite
+  depends_on "cmake" => :build
```
Comment (DomT4) [inline Library/Formula/fltk.rb]: If this is only needed for the `head` build it should be inside the `head do` block.

File: Library/Formula/fltk.rb [hunk 3]
```diff
@@ -11,27 +9,42 @@ class Fltk < Formula
...
+    depends_on MaximumMacOSRequirement => :yosemite
+    url "https://fossies.org/linux/misc/fltk-1.3.3-source.tar.gz"
+    sha256 "f8398d98d7221d40e77bc7b19e761adaf2f1ef8bb0c30eceb7beb4f2273d0d97"
+    
+    # Fixes issue with -lpng not found.
+    # Based on: https://trac.macports.org/browser/trunk/dports/aqua/fltk/files/patch-src-Makefile.diff
+    patch :DATA
+  end
+
+  # Fltk 1.3.4 include support for OS X El Capitan
```
Comment (DomT4) [inline Library/Formula/fltk.rb]: Don't need this note given the one further up.

**Comment by DomT4** [comment]
```
fltk:
 * 4: Trailing whitespace was found
 * 17: Trailing whitespace was found
```

**Comment by DomT4** [comment]
Still some whitespacing issues here:

```
        Error: 3 problems in 1 formula
fltk:
 * 4: Trailing whitespace was found
 * 15: Trailing whitespace was found
 * 19: Trailing whitespace was found
```

**Comment by DomT4** [comment]
I can't seem to clone it non-interactively locally:

````
==> Cloning http://seriss.com/public/fltk/fltk/branches/branch-1.3/
svn checkout http://seriss.com/public/fltk/fltk/branches/branch-1.3/ /usr/local/var/homebrew/cache/fltk--svn-HEAD
Authentication realm: <http://seriss.com:80> Subversion repository
Password for '':

Authentication realm: <http://seriss.com:80> Subversion repository
Username:
Password for '':
```

:cry:. We don't do interactivity as it's bad UX and means automated scripts/builds/etc fall over.

**Comment by jmtatsch** [comment]
That is weird, I don't need to use interactive mode. For me it works just fine and i don't have any creds for their svn server. I can't browse their svn via http though.
Can brew checkout from svn "natively" or is an svn installation necessary as requirement? Can the builedbot check it out?

```
brew install fltk --HEAD
==> Cloning http://seriss.com/public/fltk/fltk/branches/branch-1.3/
Updating /Library/Caches/Homebrew/fltk--svn-HEAD
==> cmake . -DCMAKE_C_FLAGS_RELEASE= -DCMAKE_CXX_FLAGS_RELEASE= -DCMAKE_INSTALL_PREFIX=/usr/local/Cellar/fltk/HEAD -DCMAKE_BUILD_TYPE=Release -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_VERB
==> make install
🍺  /usr/local/Cellar/fltk/HEAD: 175 files, 4,5M, built in 84 seconds
```

**Comment by dpo** [comment]
Just put this together to build the devel snapshot: https://gist.github.com/5b906b02e585d6f43eec
They advertise that it builds on 10.11. I don't have that system to try.

**Comment by jmtatsch** [comment]
Just checked out the formula @dpo provided. It builds fine on El Cap, doesn't suffer from that strange svn credential problem, but does configure threads and shared properly. Yay!

**Comment by dpo** [comment]
Note that there seems to be a mirror of the source on Github (https://github.com/Homebrew/homebrew-science/issues/2871#issuecomment-147500027) but I'm not sure how "official" it is.

**Comment by DomT4** [comment]
> Just put this together to build the devel snapshot: https://gist.github.com/5b906b02e585d6f43eec
They advertise that it builds on 10.11. I don't have that system to try.

Would be happy to take a `devel` build like this, but the upstream website gives the impression it would need daily updating for the URL to keep working. If that's not the case, happy to take it.

**Comment by dpo** [comment]
I just tried downloaded a few past "devel" releases, and it seems like they stick around, at least for a while. I'll prepare a PR.

**Pull request closed without merging**